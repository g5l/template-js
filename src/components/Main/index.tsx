import styles from './styles.module.css'

const Main = ({
  title = 'Typescript template',
  description = 'TypeScript, ReactJS, NextJS e CSS modules'
}) => (
  <main className={styles.wrapper}>
    <h1 className={styles.title}>{title}</h1>
    <h2 className={styles.description}>{description}</h2>
    <img
      className={styles.illustration}
      src="/img/hero-illustration.svg"
      alt="A developer"
    />
  </main>
)

export default Main
