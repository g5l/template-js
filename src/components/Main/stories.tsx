import { Meta, StoryObj } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Learning react',
    description: 'advanced react'
  },
  parameters: {
    layout: 'fullscreen'
  }
} as Meta

export const Default: StoryObj = {}

export const Basic: StoryObj = {
  args: {
    title: 'Main',
    description: 'desc'
  }
}
