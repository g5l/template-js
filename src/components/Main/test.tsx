import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    // render o component
    const { container } = render(<Main />)

    // Find the element and verify if it exists
    expect(
      screen.getByRole('heading', { name: /typescript template/i })
    ).toBeInTheDocument()

    // Generate snapshot
    expect(container.firstChild).toMatchSnapshot()
  })
})
