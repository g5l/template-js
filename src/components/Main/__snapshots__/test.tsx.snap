// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main>
  <h1>
    Typescript template
  </h1>
  <h2>
    TypeScript, ReactJS, NextJS e CSS modules
  </h2>
  <img
    alt="A developer"
    src="/img/hero-illustration.svg"
  />
</main>
`;
